homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,620 @@
+                },
+            )
+        }
+    )
+
+
+async def _get_flow_step_for_editing(
+    user_input: dict[str, Any],
+) -> str:
+    """Choose which observation config flow form step to show depending on observation type selected."""
+
+    observations: list[dict[str, Any]] = user_input[CONF_OBSERVATIONS]
+    selected_idx = int(user_input[CONF_INDEX])
+
+    return str(observations[selected_idx][CONF_PLATFORM])
+
+
+async def _get_state_schema(
+    handler: SchemaCommonFlowHandler,
+) -> vol.Schema:
+    """Return the state schema, without an add_another box if editing or using the schema for previews."""
+
+    if not hasattr(handler, "options") or handler.options.get(CONF_INDEX) is None:
+        return STATE_SUBSCHEMA.extend(ADD_ANOTHER_BOX_SCHEMA.schema)
+
+    return STATE_SUBSCHEMA
+
+
+async def _get_numeric_state_schema(
+    handler: SchemaCommonFlowHandler,
+) -> vol.Schema:
+    """Return the numeric_state schema, without an add_another box if editing or using the schema for previews."""
+
+    if not hasattr(handler, "options") or handler.options.get(CONF_INDEX) is None:
+        return NUMERIC_STATE_SUBSCHEMA.extend(ADD_ANOTHER_BOX_SCHEMA.schema)
+
+    return NUMERIC_STATE_SUBSCHEMA
+
+
+async def _get_template_schema(
+    handler: SchemaCommonFlowHandler,
+) -> vol.Schema:
+    """Return the template schema, without an add_another box if editing or using the schema for previews."""
+
+    if not hasattr(handler, "options") or handler.options.get(CONF_INDEX) is None:
+        return TEMPLATE_SUBSCHEMA.extend(ADD_ANOTHER_BOX_SCHEMA.schema)
+
+    return TEMPLATE_SUBSCHEMA
+
+
+async def _add_more_or_end(
+    user_input: dict[str, Any],
+) -> ConfigFlowSteps | None:
+    """Choose whether to add another observation or end the flow."""
+    if user_input.get("add_another", False):
+        return ConfigFlowSteps.OBSERVATION_SELECTOR
+    return None
+
+
+async def _get_base_suggested_values(
+    handler: SchemaCommonFlowHandler,
+) -> dict[str, Any]:
+    """Return suggested values for the base sensor options."""
+
+    return _convert_fractions_to_percentages(dict(handler.options))
+
+
+async def _get_observation_values_if_editing(
+    handler: SchemaCommonFlowHandler,
+) -> dict[str, Any]:
+    """Only if editing observations in options flow, get the values. Otherwise leave blank."""
+    if idx := handler.options.get(CONF_INDEX):
+        return _convert_fractions_to_percentages(
+            dict(handler.options[CONF_OBSERVATIONS][int(idx)])
+        )
+    return {}
+
+
+async def _validate_user(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate user input for the basic settings and convert to fractions for storage."""
+    if user_input[CONF_PRIOR] == 0:
+        raise SchemaFlowError("prior_low_error")
+    if user_input[CONF_PRIOR] == 100:
+        raise SchemaFlowError("prior_high_error")
+    if user_input[CONF_PROBABILITY_THRESHOLD] == 0:
```

Comment:
Why does the schema allow to set `0` if we are anyhow going to say it's too low?
Likewise for 100.

Better to set the limits correctly in the schema than to raise errors after.

Reply:
Because whilst 0 is too low there is no non zero value that is too low. 0.00001, 0.000000001 etc are all valid and plausible values.
Ideally the schema would just define that `0 < input < 100`. However the constraint is provided by the [NumberSelector](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/selector.py#L1055) class which takes only `min` and `max` values which accept inputs equal to those values (i.e they are inclusive).

Reply:
I see that vol.Range includes an option to be non-inclusive
https://github.com/alecthomas/voluptuous/blob/a7a55f83b9fa7ba68b0669b3d78a61de703e0a16/voluptuous/validators.py#L643
I will use that.

Reply:
The user can anyhow not set 0.0001 as the steps are in wholes and therefore 1,2,3,4...

Reply:
The set only effects the slider, they can still input numbers into the box that have decimals etc.
I now have working code using vol.Range() instead, I'm trying to see if I can retain some of the verbosity of the custom errors of `threshold_low_error` and `prior_low_error` as they help to 'teach' the user more about why those values are not allowed.

Reply:
When I tested yesterday it wasn't possible to put in decimals when the `step` didn't allow it but that was in a box and not a slider so maybe that's different (or it allows it but still shows red).
Anyhow good if you find an alternative which works.

Reply:
Implemented in a3cd9e4